# Week 2 Assignment - Correlation Analysis Assignment

## Objective: Write Python to analyze monthly 2024 data for Temperature, GreenTote, and SolarTrek Water Bottle

In [8]:
# Create dictionaries for all 3 varibles using Months as the keys and each respective variable as the values

# Dictionary for temperature in °F
Temperature = {
    "January": 40,
    "February": 45,
    "March": 55,
    "April": 65,
    "May": 75,
    "June": 85,
    "July": 90,
    "August": 88,
    "September": 86,
    "October": 80,
    "November": 70,
    "December": 60
}

In [12]:
# Dictionary for GreenTote sales
GreenTote_Sales = {
    "January": 87500,
    "February": 100625,
    "March": 115725,
    "April": 132075,
    "May": 148875,
    "June": 164500,
    "July": 172725,
    "August": 185800,
    "September": 180900,
    "October": 170000,
    "November": 165000,
    "December": 160000
}

In [13]:
# Dictionary for SolarTrek Water Bottle sales (using dict() function and tuple method from lectutr)
SolarTrek_Bottle_Sales = dict([
    ("January", 80900),
    ("February", 85200),
    ("March", 92500),
    ("April", 100980),
    ("May", 110500),
    ("June", 118300),
    ("July", 124800),
    ("August", 130500),
    ("September", 128800),
    ("October", 120000),
    ("November", 114000),
    ("December", 110000)
])


In [24]:
# Finding the highest and lowest sale months for GreenTote
# Using the max() and min() funtions to get the highest and lowest sales months, and the key argument to return the respective sales value

GreenTote_Max_Sales = max(GreenTote_Sales, key=GreenTote_Sales.get) # returns the month with the highest sales
GreenTote_Min_Sales = min(GreenTote_Sales, key=GreenTote_Sales.get) # returns the month with the lowest sales

print("GreenTote - Highest Sales:")
print(f"{GreenTote_Max_Sales}: ${GreenTote_Sales[GreenTote_Max_Sales]:,}") # Prints the month with the highest sales by indexing month and sales

print("GreenTote - Lowest Sales:")
print(f"{GreenTote_Min_Sales}: ${GreenTote_Sales[GreenTote_Min_Sales]:,}") # Prints the month with the lowest sales by indexing month and sales


GreenTote - Highest Sales:
August: $185,800
GreenTote - Lowest Sales:
January: $87,500


In [23]:
# Repeat step about to find highest and lowest sale months for SolarTrek Bottles

SolarTrek_Max_Sales = max(solar_trek_bottle, key=solar_trek_bottle.get)
SolarTrek_Min_Sales = min(solar_trek_bottle, key=solar_trek_bottle.get)

print("SolarTrek Water Bottle - Highest Sales:")
print(f"{SolarTrek_Max_Sales}: ${solar_trek_bottle[SolarTrek_Max_Sales]:,}")

print("SolarTrek Water Bottle - Lowest Sales:")
print(f"{SolarTrek_Min_Sales}: ${SolarTrek_Bottle_Sales[SolarTrek_Min_Sales]:,}")

SolarTrek Water Bottle - Highest Sales:
August: $130,500
SolarTrek Water Bottle - Lowest Sales:
January: $80,900


In [31]:
# Repeat the same methodolology to find the highest and lowest temperature months

Max_Temp = max(Temperature, key=Temperature.get) # returns month with the highest temperature
Min_Temp = min(Temperature, key=Temperature.get) # returns month with the lowest temperature

print("Highest Temperature:")
print(f"{Max_Temp}: {Temperature[Max_Temp]}°F") # Prints the month with the highest temperature by indexing month and temp

print("Lowest Temperature:")
print(f"{Min_Temp}: {Temperature[Min_Temp]}°F") # Prints the month with the lowest temperature by indexing month and temp

Highest Temperature:
July: 90°F
Lowest Temperature:
January: 40°F


In [32]:
# Calculate Pearson correlation coefficent to determine correlation between dictionaries

In [33]:
# Import neccessary package
import scipy.stats

In [37]:
# Convert dictionaries to lists
GreenTote_Sales_list = list(GreenTote_Sales.values())
SolarTrek_Sales_list = list(SolarTrek_Bottle_Sales.values())
Temperature_list = list(Temperature.values())

In [45]:
# Now compare Correlation between GreenTote sales vs. SolarTrek Water Bottle Sales

Test1 = scipy.stats.pearsonr(GreenTote_Sales_list, SolarTrek_Sales_list)

In [46]:
# Compare Correlation between Temperature vs. GreenTote Sales
Test2 = scipy.stats.pearsonr(Temperature_list, GreenTote_Sales_list)

In [44]:
# Compare Temperature vs. SolarTrek Water Bottle Sales
Test3 = scipy.stats.pearsonr(Temperature_list, SolarTrek_Sales_list)

In [55]:
# Print Resuts
print("Correlation between GreenTote sales vs. SolarTrek Water Bottle Sales:")
print(Test1)

Correlation between GreenTote sales vs. SolarTrek Water Bottle Sales:
PearsonRResult(statistic=0.9880734176209012, pvalue=1.8628514007033096e-09)


In [56]:
print("Correlation between Temperature vs. GreenTote Sales:")
print(Test2)

Correlation between Temperature vs. GreenTote Sales:
PearsonRResult(statistic=0.9249120857886273, pvalue=1.6556171942382138e-05)


In [58]:
print("Correlation between Temperature vs. SolarTrek Water Bottle Sales:")
print(Test3)

Correlation between Temperature vs. SolarTrek Water Bottle Sales:
PearsonRResult(statistic=0.9610765652591183, pvalue=6.590554270244605e-07)


# OR

<div class="alert alert-block alert-success">
Correlation between GreenTote sales vs. SolarTrek Water Bottle Sales:<br>
PearsonRResult(statistic=0.9880734176209012, pvalue=1.8628514007033096e-09)
</div>

<div class="alert alert-block alert-success">
Correlation between Temperature vs. GreenTote Sales:<br>
PearsonRResult(statistic=0.9249120857886273, pvalue=1.6556171942382138e-05)
</div>

<div class="alert alert-block alert-success">
Correlation between Temperature vs. SolarTrek Water Bottle Sales: <br>
PearsonRResult(statistic=0.9610765652591183, pvalue=6.590554270244605e-07)
</div>

### Based on the results, I am opting to take Option 1 and develop and joint marketing campaign for the 2 products. <br>
### The nearly 1 statistic value and the very low p-value, tells me that the 2 products are highly correlated - when one's sales go up so do the other's. In addition, they both have high statistic values and low p-values with temperature. They both respond strongly to temperature, warmer weather brings in higher sales for both products indicating seasonal demand.